mixer/adapter/stackdriver/metric/bufferedClient.go

```diff
@@ -74,19 +88,23 @@ func (b *buffered) Send() {
 	b.m.Unlock()
 
 	merged := merge(toSend, b.l)
-	err := b.pushMetrics(context.Background(),
-		&monitoringpb.CreateTimeSeriesRequest{
-			Name:       fmt.Sprintf("projects/%s", b.project),
-			TimeSeries: merged,
-		})
-
-	// TODO: this is executed in a daemon, so we can't get out info about errors other than logging.
-	// We need to build framework level support for these kinds of async tasks. Perhaps a generic batching adapter
-	// can handle some of this complexity?
-	if err != nil {
-		_ = b.l.Errorf("Stackdriver returned: %v\nGiven data: %v", err, merged)
-	} else {
-		b.l.Debugf("Successfully sent data to Stackdriver.")
+	batches := batchTimeSeries(merged, b.timeSeriesBatchSize)
+
+	for _, timeSeries := range batches {
+		err := b.pushMetrics(context.Background(),
+			&monitoringpb.CreateTimeSeriesRequest{
+				Name:       fmt.Sprintf("projects/%s", b.project),
```

Comment:
Instead of Sprintf, you can just use concatenation which is considerably more efficient.

Reply:
Thanks for reviewing! Done.